app/components/events/view/export/api-response.js

```diff
@@ -1,16 +1,61 @@
 import Component from '@ember/component';
-import { computed } from '@ember/object';
+import fetch from 'fetch';
+import { computed, observer } from '@ember/object';
 import { htmlSafe } from '@ember/string';
-
+import ENV from 'open-event-frontend/config/environment';
 
 export default Component.extend({
-  json: computed(function() {
-    return htmlSafe(syntaxHighlight(
-      '{\n "name": "Sample event",\n "starts_at" : "2017-03-10T17:00:0",\n "ends_at":"2017-03-12T17:00:0"\n}'
-    ));
-  })
+
+  didInsertElement() {
+    this._super(...arguments);
+    fetch(this.get('displayUrl'))
+      .then(res => res.json())
+      .then(json => {
+        json = JSON.stringify(json, null, 2);
+        this.set('json', htmlSafe(syntaxHighlight(json)));
+      });
+  },
+
+  baseUrl: computed(function() {
+    return `${`${ENV.APP.apiHost}/${ENV.APP.apiNamespace}/events/`}${this.get('eventID')}`;
+  }),
+
+  displayUrl: computed(function() {
+    return `${`${ENV.APP.apiHost}/${ENV.APP.apiNamespace}/events/`}${this.get('eventID')}`;
+  }),
+
+  eventIDObserver: observer('eventID', () => {
+    resetURL(this.get('baseUrl'), this.get('eventID'));
+    resetURL(this.get('displayUrl'), this.get('eventID'));
+  }),
+
+  toggleSwitches: {
+    sessions       : false,
+    microlocations : false,
+    tracks         : false,
+    speakers       : false,
+    sponsors       : false,
+    tickets        : false
+  },
+
+  actions: {
+    checkboxChange(data) {
+      this.set(`toggleSwitches.${data}`, !this.get(`toggleSwitches.${data}`));
```

Comment:
Group both of this consecutive `set` statements into one `setProperties` statement so that both run within a single runloop